src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java

```diff
@@ -0,0 +1,79 @@
+package io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs;
+
+import com.datastax.oss.driver.api.core.data.CqlDuration;
+
+/**
+ * Helper class for converting {@link CqlDuration} to ISO-8601 duration string ("PnYnMnDTnHnMnS"),
+ * translation not supported by {@link CqlDuration} itself even tho it can parse such strings.
+ */
+abstract class CqlDurationConverter {
+  private static final long NANOS_PER_SECOND = 1_000_000_000L;
+  private static final long NANOS_PER_MINUTE = 60 * NANOS_PER_SECOND;
+  private static final long NANOS_PER_HOUR = 60 * NANOS_PER_MINUTE;
+
+  /**
+   * Convert {@link CqlDuration} to ISO-8601 duration string ("PnYnMnDTnHnMnS").
+   *
+   * @param value CqlDuration to convert to ISO-8601 duration string
+   * @return ISO-8601 compliant duration string
+   */
+  public static String toISO8601Duration(CqlDuration value) {
+    int months = value.getMonths();
+    int days = value.getDays();
+    long nanoSeconds = value.getNanoseconds();
+
+    final StringBuilder sb = new StringBuilder("P");
+
+    // Do we have date part?
+    if (months > 0 || days > 0) { // yes
+      final int years = months / 12;
+      months = months % 12;
+
+      if (years > 0) {
+        sb.append(years).append("Y");
+      }
+      if (months > 0) {
+        sb.append(months).append("M");
+      }
+      if (days > 0) {
+        sb.append(days).append("D");
+      }
+    }
+    // Check if we have time part
+    if (nanoSeconds == 0L) {
+      if (sb.length() == 1) {
+        return "PT0S"; // All zeroes case
+      }
+      return sb.toString(); // Only date part exists
+    }
+
+    sb.append('T');
+    long hours = nanoSeconds / NANOS_PER_HOUR;
```

Comment:
Neat trick, looks like dividing 2 longs gives you the value rounded down :+1:

Reply:
Yes, in that's the logic in Java (and other C-derivative languages) wrt integer division.